src/chessboard/components/Square.tsx

```diff
@@ -150,7 +150,10 @@ export function Square({
       }}
       onDragEnter={() => onDragOverSquare(square)}
       onClick={() => {
-        onSquareClick(square);
+        const piece = Array.isArray(children)
+          ? (children.find((c) => c?.props?.piece)?.props?.piece as Piece)
+          : undefined;
```

Comment:
get piece by square is much easier 
```suggestion
        const piece = currentPosition[square]
```

Reply:
wow. Yeah didn't realise that was there! Much simpler